test/index.test-d.ts

```diff
@@ -109,10 +123,10 @@ const runServer = async () => {
   app.post('/upload/raw/any', async function (req, reply) {
     const parts = req.parts()
     for await (const part of parts) {
-      if (part.file) {
+      if ('file' in part) {
```

Comment:
This should be modified on README as well. But this needs to be revisited to be on the type.

Reply:
Your problem is `TypeScript` issue.
`TypeScript` cannot restrict the type in certain format that's the cause.

The `README` example is certainly valid in `Javascript`.

Reply:
Yes, typescript wants you to use `in` for narrowing in this situation, which works for both JS and TS.
https://www.typescriptlang.org/docs/handbook/2/narrowing.html#the-in-operator-narrowing

A discriminator would perhaps be ideal:
```ts
export interface MultipartFile {
  type: 'file',
  ...
}
export interface MultipartValue<T = unknown> {
  type: 'value',
  ...
}

if (part.type === 'file') {
  part.toBuffer()
}
```